src/index.ts

```diff
@@ -1583,8 +1578,16 @@ export class BigQuery extends common.Service {
       }
       // The Job is important for the `queryAsStream_` method, so a new query
       // isn't created each time results are polled for.
+
       options = extend({job}, options);
-      job!.getQueryResults(options, callback as QueryRowsCallback);
+      const datasetId = job!.metadata.configuration.query.destinationTable
+        .datasetId;
+      const tableId = job!.metadata.configuration.query.destinationTable
+        .tableId;
+      const dataset = this.dataset(datasetId);
+      const table = dataset.table(tableId);
+
+      table.getRows(options, callback as RowsCallback);
```

Comment:
I think this is my mistake, it looks like `getQueryResults` has some polling logic included. I think we just need to do something like

```js
job
  .on('error', err => callback(err))
  .on('complete', () => {
    table.getRows(options, callback);
  });

```

Does that sound right @stephenplusplus?

Reply:
Yes, that sounds right!

Reply:
Thanks! Made this update, still have a failing system-test that I can't locate the fix for.

Reply:
Ah, looks like maybe list tabledata won't work for certain queries (like creating models). I'm kind of surprised it returns a table reference at all in this case. We may need to conditionally call `getRows` and `getQueryResults` depending on the type of query.